### DISCUSSION TITLE
preload insdie load function

### DISCUSSION THREAD

**Discussion by mkvcc** [adonisjs/core#2922] [Help]
Hi.
Is it possible to do something like this:
`
user.load('posts' , (q) => {
 q.preload('comments') ;
})
`
I'm currently getting sql error with this code.
thanks in advance

**Comment by thetutlage** [comment]
You haven't shared the SQL error

**Reply by mkvcc** [reply]
`
select [prev_person_artists].*, [prev_opus_groups].[opus] as [pivot_opus], [prev_opus_groups].[artist] as [pivot_artist] from [prev_person_artists] inner join [prev_opus_groups] on [prev_person_artists].[id] = [prev_opus_groups].[artist] where [prev_opus_groups].[opus] in ((@p0, @p1), (@p2, @p3), (@p4, @p5), (@p6, @p7), (@p8, @p9), (@p10, @p11), (@p12, @p13), (@p14, @p15), (@p16, @p17), (@p18, @p19), (@p20, @p21), (@p22, @p23), (@p24, @p25), (@p26, @p27), (@p28, @p29), (@p30, @p31), (@p32, @p33), (@p34, @p35), (@p36, @p37), (@p38, @p39), (@p40, @p41), (@p42, @p43), (@p44, @p45), (@p46, @p47), (@p48, @p49), (@p50, @p51), (@p52, @p53), (@p54, @p55), (@p56, @p57), (@p58, @p59), (@p60, @p61), (@p62, @p63), (@p64, @p65), (@p66, @p67), (@p68, @p69), (@p70, @p71), (@p72, @p73), (@p74, @p75), (@p76, @p77), (@p78, @p79), (@p80, @p81), (@p82, @p83), (@p84, @p85), (@p86, @p87), (@p88, @p89), (@p90, @p91), (@p92, @p93), (@p94, @p95), (@p96, @p97), (@p98, @p99), (@p100, @p101), (@p102, @p103), (@p104, @p105), (@p106, @p107), (@p108, @p109), (@p110, @p111), (@p112, @p113), (@p114, @p115), (@p116, @p117), (@p118, @p119), (@p120, @p121), (@p122, @p123), (@p124, @p125), (@p126, @p127), (@p128, @p129), (@p130, @p131), (@p132, @p133), (@p134, @p135), (@p136, @p137), (@p138, @p139), (@p140, @p141), (@p142, @p143), (@p144, @p145), (@p146, @p147), (@p148, @p149), (@p150, @p151), (@p152, @p153), (@p154, @p155), (@p156, @p157), (@p158, @p159), (@p160, @p161), (@p162, @p163), (@p164, @p165), (@p166, @p167), (@p168, @p169), (@p170, @p171), (@p172, @p173), (@p174, @p175), (@p176, @p177), (@p178, @p179), (@p180, @p181), (@p182, @p183)) 
`

artist and opus have many to many relation with each other with artist and opus foreign key.
and i'm loading opus and artist group.


`
artist.load('opuses' , (q) => {
 q.preload('artists' , (q) => { q.where('x' , 'y') ; } 
}) 
`

and its running on sql server database.

**Reply by RomainLanz** [reply]
There is no error in the block you shared.

**Reply by mkvcc** [reply]
` Incorrect syntax near ','`
just an sql error

**Comment by WeyIin** [comment]
Can you provide us with model and relationship and migration files

**Reply by mkvcc** [reply]
sorry I can't. I don't think company policies allows it. 
but its a pretty simple relation. I will test this is a different project and post the result as soon as I can. If it works then there is something wrong in my project.

relation: 
simple many to many relation between User and Post.

and the query should be:
`let user = await User.first() ;`
`
await user.load('posts' , (q) => {
  q.preload('users' , (user_q) => {
    u_query.where('role' , 'x') 
  }) ;
})
`

**Reply by WeyIin** [reply]
Then at least mimic and share dummy code and data.

I use latest AdonisJS version and many to many works fine.
So there must be an issue with your stuff and / or data.

Edit: when making dummy code and data make sure you can replicate your own issue first :)
